src/lib/users/clients/inst_root_first.rb

```diff
@@ -0,0 +1,274 @@
+        "is recommended. To ensure that the password was entered correctly,\n" \
+        "reenter it in a second field.\n" \
+        "</p>\n"
+      ) <<
+
+      # help text, continued 3
+      _(
+        "<p>\n" \
+        "All the rules for user passwords apply to the \"root\" password:\n" \
+        "Distinguish between uppercase and lowercase. A password should have at\n" \
+        "least 5 characters and, as a rule, not contain any accented letters or umlauts.\n" \
+        "</p>\n"
+      )
+
+      helptext = helptext + UsersSimple.ValidPasswordHelptext
+
+      # help text, continued 4
+      helptext << _(
+        "<p>\n" \
+        "Do not forget this \"root\" password.\n" \
+        "</p>"
+      )
+
+      if UsersUtils.check_ca_constraints?
+        helptext << Builtins.sformat(
+          # additional help text about password
+          _(
+            "<p>If you intend to use this password for creating certificates,\n" +
+            "it has to be at least %1 characters long.</p>"
+          ),
+          UsersUtilsClass::MIN_PASSWORD_LENGTH_CA
+        )
+      end
+
+      helptext
+    end
+
+    # Sets the wizard dialog contents
+    def create_ui
+      Wizard.SetTitleIcon("yast-users")
+
+      Wizard.SetContents(
+        # Title for root-password dialogue
+        _("Password for the System Administrator \"root\""),
+        root_password_ui,
+        root_password_help,
+        GetInstArgs.enable_back || Mode.normal,
+        GetInstArgs.enable_next || Mode.normal
+      )
+    end
+
+    # Handles user's input and returns symbol what to do next
+    # @return [Symbol] :next, :back or :abort
+    def handle_ui_passwords
+      begin
+        UI.SetFocus(Id(:pw1))
+        ret = Wizard.UserInput
+
+        if ret == :abort || ret == :cancel
+          if Popup.ConfirmAbort(:incomplete)
+            ret = :abort
+          else
+            ret = :try_again
+            next
+          end
+        end
+
+        ret = :next if ret == :accept # from proposal
+
+        if ret == :next
+          password_1 = Convert.to_string(UI.QueryWidget(Id(:pw1), :Value))
+          password_2 = Convert.to_string(UI.QueryWidget(Id(:pw2), :Value))
+
+          if validate_password(password_1, password_2)
+            UsersSimple.SetRootPassword(password_1)
+          else
+            ret = :try_again
+          end
+        end
+      end until ret == :next || ret == :back || ret == :abort
+
+      ret
+    end
+
+    # UI handler function, loads and unloads cracklib
+    def handle_ui
+      unless UsersSimple.LoadCracklib
+        log.errror "loading cracklib failed, not used for pw check"
+        UsersSimple.UseCrackLib(false)
+      end
+
+      begin
+        return handle_ui_passwords
+      ensure
+        UsersSimple.UnLoadCracklib
+      end
+    end
+
+    # Validates whether password1 and password2 match and are valid
+    def validate_password(password_1, password_2)
```

Comment:
Sounds like a good idea, on the other hand, having `password_1` and `password_2` will be rather an exception in using this new class.